lib/plan-finder.js

```diff
@@ -152,22 +157,37 @@ class PlanFinder {
                 * @param {String} name transformer name
                 * @param {Object} attributes custom attributes for this step
      */
-    formPlan(source, outputInstructions, currentPlan) {
+    async formPlan(source, outputInstructions, currentPlan) {
         if (!currentPlan || currentPlan.length === 0) {
             debug(`No plan could be found for : ${source.type} -> ${outputInstructions.type}`);
             throw new RenditionFormatUnsupportedError(`No plan could be found for : ${source.type} -> ${outputInstructions.type}`);
         }
+
         // userData is an important field for sending IO events in Nui
         // for now, add them to every output step instructions
         // future plan: store these somewhere, maybe in the plan or engine level to put inside every event
         const userData = outputInstructions.userData;
-
         const plan = [];
+
+        // get image metadata and check for "Orientation" property if this is for Sensei
+        if (this.registeredTransformers.length === 1 && this.registeredTransformers[0].includes("SenseiTransformer")) {
```

Comment:
```suggestion
        if (currentPlan.length === 1 && currentPlan[0].includes("SenseiTransformer")) {
```
the idea is if there is only one transformer in the plan determined by bfs plan, and the transformer is a sensei transformer, this means we need to check the orientation of the source file since it won't be auto-applied

Reply:
Makes sense now ;).